spec/controllers/orders_controller_spec.rb

```diff
@@ -683,46 +680,70 @@ def deliver; end
 
-        it "sets the actual times to the completed reservation times" do
-          do_request
-          expect(@reservation.reload.actual_start_at).to match_date @reservation.reserve_start_at
-          expect(@reservation.actual_end_at).to match_date(@reservation.reserve_start_at + 60.minutes)
-        end
+        context "for a non reservation-only instrument" do
+          before :each do
+            @instrument = FactoryGirl.create(:instrument,
+                                             facility: @authable,
+                                             facility_account: @facility_account)
+            setup_past_reservation
+          end
 
-        it "assigns a price policy and cost" do
-          do_request
-          expect(@order_detail.reload.price_policy).not_to be_nil
-          expect(@order_detail.actual_cost).not_to be_nil
-        end
+          it "sets the order_detail states to completed because it's in the past" do
+            do_request
+            assigns[:order].order_details.all? { |od| expect(od.state).to eq("complete") }
+          end
 
-        context "canceled" do
-          before :each do
-            @params[:order_status_id] = OrderStatus.canceled.first.id
+          it "sets the order_detail fulfilment dates to the order time" do
             do_request
+            assigns[:order].order_details.all? do |od|
+              expect(od.fulfilled_at).not_to be_nil
+              expect(od.fulfilled_at).to match_date @reservation.reserve_end_at
+            end
           end
 
-          it "is able to set order_detail states to canceled" do
-            assigns[:order].order_details.all? { |od| expect(od.state).to eq("canceled") }
+          it "sets the actual times to the completed reservation times" do
+            do_request
+            expect(@reservation.reload.actual_start_at).to match_date @reservation.reserve_start_at
+            expect(@reservation.actual_end_at).to match_date(@reservation.reserve_start_at + 60.minutes)
+          end
+
+          it "assigns a price policy and cost" do
+            do_request
+            expect(@order_detail.reload.price_policy).not_to be_nil
+            expect(@order_detail.actual_cost).not_to be_nil
           end
 
-          it "sets the canceled time on the reservation" do
-            assigns[:order].order_details.all? { |od| expect(od.reservation.canceled_at).not_to be_nil }
-            expect(@reservation.reload.canceled_at).not_to be_nil
-            # Should this match the date put in the form, or the date when the action took place
-            # @reservation.canceled_at.should match_date @submitted_date
+          context "canceled" do
+            before :each do
+              @params[:order_status_id] = OrderStatus.canceled.first.id
+              do_request
+            end
+
+            it "is able to set order_detail states to canceled" do
+              assigns[:order].order_details.all? { |od| expect(od.state).to eq("canceled") }
```

Comment:
As above, would this work? `expect(assigns[:order].order_details).to all(be_canceled)`